crates/oxc_transformer/src/es2019/optional_catch_binding.rs

```diff
@@ -58,8 +58,14 @@ impl<'a> Traverse<'a> for OptionalCatchBinding<'a> {
         if clause.param.is_some() {
             return;
         }
-        let symbol_id =
-            ctx.generate_uid("unused", ctx.scoping.current_scope_id(), SymbolFlags::CatchVariable);
+
+        // Get the BlockStatement scope_id below the CatchClause scope
+        let child_scope_id = ctx.scopes().get_child_ids(clause.scope_id.get().unwrap())[0];
```

Comment:
I think this is simpler, and also avoid a lookup in `ctx.scopes()`:

```suggestion
        let child_scope_id = clause.body.scope_id.get().unwrap();
```

Also, I have been wondering if we can remove `child_ids` from `ScopeTree` - it's rarely used, and you can also always get child scopes by visiting from the node (like `ChildScopeCollector` does), so it's not necessary to have it. I suspect the cost of building and maintaining `child_ids` is not worth it.

Reply:
Also, I'd suggest renaming `child_scope_id` var here to `body_scope_id` - that's more descriptive.

Reply:
Why didn't I realize I could get the scope_id from inside `clause.body`!